multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/ProcessConflictPreventer.java

```diff
@@ -70,7 +81,7 @@ private List<Operation> findConflictingOperations(String mtaId, String namespace
                                .mtaId(mtaId)
                                .namespace(namespace)
                                .spaceId(spaceId)
-                               .acquiredLock(true)
+                               .orderByStartTime(OrderDirection.DESCENDING)
```

Comment:
Why acquired lock check is done now in the code?

Reply:
If one process hangs prior to the step that sets this lock and another process is started in that time, there would be 2 processes with `hasAcquiredLock` -> false. The current query didn't handle that scenario, that's why we need to manually check the lock.